Automattic/woocommerce-payments
# Migrate DetailsLink component to TS to improve code quality

Part of #6929

#### Changes proposed in this Pull Request

This PR refactors the `DetailsLink` component and the shared function `getDetailsURL` to TypeScript to improve code quality and enhance developer experience.

#### Testing instructions





* Run `npx tsc --noEmit` and ensure no errors are returned.
* Ensure all tests pass with no changes to snapshot tests.

-------------------


**Post merge**
